src/chrdev.rs

```diff
@@ -0,0 +1,42 @@
+use bindings;
+use c_types;
+use error;
+
+pub struct DeviceNumberRegion {
+    dev: bindings::dev_t,
+    count: usize,
+}
+
+impl DeviceNumberRegion {
+    pub fn allocate(
+        count: usize,
+        first_minor: usize,
```

Comment:
I wonder if it's worth using `Range` here to make it more readable? `dev = allocate(0..1, "chrdev-tests\0")` Mostly having multiple function arguments of the same type seems like a thing you do in a language without a great typesystem :)

If there were some `IntoRange` trait or something so I could pass either a single `usize` or ` Range<usize>` then I think it'd be real compelling.